### DISCUSSION TITLE
[Question][Hw6] Is there a way to handle unicode without external library?

### DISCUSSION THREAD

**Discussion by m-spitfire** [prosyslab-classroom/cs348-information-security#722] [Q&A]
Name: Murad Bashirov

Hello. I'm trying to generate a character from unicode codepoint as described in the json specification: 
![image](https://github.com/prosyslab-classroom/cs348-information-security/assets/66242799/e436d78d-9520-48b0-8933-fdc3056a3c50)

But from my search in ocaml you have to use an external library like [uutf](https://github.com/dbuenzli/uutf) to do this, is there a way to do without an external library?

**Comment by m-spitfire** [comment]
oops, sorry for notification, just `"\u{codepoint}"` works...

**Comment by bongjunj** [comment]
First code point | Last code point | Byte 1 | Byte 2 | Byte 3 | Byte 4
-- | -- | -- | -- | -- | --
U+0000 | U+007F | 0xxxxxxx |  
U+0080 | U+07FF | 110xxxxx | 10xxxxxx |  
U+0800 | U+FFFF | 1110xxxx | 10xxxxxx | 10xxxxxx |  
U+10000 | U+10FFFF | 11110xxx | 10xxxxxx | 10xxxxxx | 10xxxxxx

```
character
    '0020' . '10FFFF' - '"' - '\'
    '\' escape
```

You can implement a function that converts a code point into a stream of bytes, and call `Bytes.to_string`.


Reference: https://en.wikipedia.org/wiki/UTF-8#Encoding

**Comment by m-spitfire** [comment]
Thanks @bonjune!
